src/tensor_operations/matrix_algebra.jl

```diff
@@ -86,15 +88,7 @@ function exp(A::ITensor; kwargs...)
   return exp(A, Lis, Ris; kwargs...)
 end
 
-function map_diag!(f::Function, it_destination::ITensor, it_source::ITensor)
+function NDTensors.map_diag!(f::Function, it_destination::ITensor, it_source::ITensor)
   return itensor(map_diag!(f, tensor(it_destination), tensor(it_source)))
 end
 map_diag(f::Function, it::ITensor) = map_diag!(f, copy(it), it)
-
-function map_diag!(f::Function, t_destination::Tensor, t_source::Tensor)
-  for i in 1:diaglength(t_destination)
-    NDTensors.setdiagindex!(t_destination, f(NDTensors.getdiagindex(t_source, i)), i)
-  end
-  return t_destination
-end
-map_diag(f::Function, t::Tensor) = map_diag!(f, copy(t), t)
```

Comment:
In order to make CI happy and also for a smoother upgrade path for users, we could leave ITensors.jl as-is in this PR (i.e. don't remove `ITensors.map_diag!` and don't use `NDTensors.map_diag!` in ITensors.jl yet). Then we can make a follow-up PR where we make these change to ITensors.jl, after a new version of NDTensors.jl is registered that defines `NDTensors.map_diag!`.

Reply:
Okay i added these definitions back and removed the `using NDTensor: map_diag!` from this file for now